src/dapp/components/ui/HalveningBanner.tsx

```diff
@@ -1,24 +1,56 @@
+import "./Banner.css";
+
+import { useService } from "@xstate/react";
 import React, { useState } from "react";
 
-import alert from "../../images/ui/expression_alerted.png";
 import closeIcon from "../../images/ui/close.png";
-
-import "./Banner.css";
+import alert from "../../images/ui/expression_alerted.png";
+import { service } from "../../machine";
+import { numberWithBreaks } from "../../utils/number";
+import { getNextHalvingThreshold } from "../../utils/supply";
 
 export const Banner: React.FC = () => {
   const [show, setShow] = useState(true);
+  const [totalSupply, setTotalSupply] = React.useState<number>(0);
+
+  const [machineState, send] = useService<
+    Context,
+    BlockchainEvent,
+    BlockchainState
+  >(service);
+
+  React.useEffect(() => {
+    const load = async () => {
+      const notReady =
```

Comment:
Consider checking `isConnected` on the `machineState.context.blockChain`.

Do you think we could use this to determine if we can show the warning or not?